### PR TITLE
Feat/improvment config history

### PR DESCRIPTION
**Pull request by Bozonych**

### CODE REVIEW

File: packages/administration/dashboards-config/injectors.ts [hunk 1]
```diff
@@ -15,23 +15,29 @@ export class DashboardConfig {
...
+export abstract class EntityCollectionWithHistory<T> extends EntityCollectionBase<T> {
+    protected httpClient: HttpClient = inject<HttpClient>(HttpClient);
+
+    constructor(factoryService: EntityCollectionFactoryService, private apiUrl: string) {
+        super(factoryService.create(apiUrl));
     }
-    public getHistoryById(id: number): Observable<AuditResponse> {
-        return this.httpClient.get<AuditResponse>(`dashboard/api/dashboards-audit/${id}`);
+
+    public getHistoryById(id: number, pagination: { page: number; size: number } = { page: 0, size: 10 }): Observable<AuditResponse> {
```
Comment (denisogolubev) [inline packages/administration/dashboards-config/injectors.ts:25]: Why don't you use the params property for the get request?
```ts
public getHistoryById(id: number, pagination: { page: number; size: number } = { page: 0, size: 10 }): Observable<AuditResponse> {
    const { page, size } = pagination;
    const params = {
      page,
      size,
      sort: 'revtstmp,desc'
    }
    return this.httpClient.get<AuditResponse>(`${this.apiUrl}-audit/${id}`, {params});
  }
```

**Review by denisogolubev — APPROVED** [review]

File: packages/administration/dashboards-config/configuration-history/config-history-modal/config-history-modal.component.html [hunk 2]
```diff
@@ -48,9 +48,10 @@ <h3 class="text-uppercase font-weight-bold">
         <mat-card class="w-100">
             <mat-card-content>
                 <h2 class="w-100">{{ 'dashboard-config-widget.change-lists' | translate }}</h2>
-                <mat-list *ngIf="data?.events?.length; else noHistory">
+                <mat-list *ngIf="data?.events?.length; else noHistory" class="d-flex flex-column justify-content-center">
+                    @for(event of data?.events; track event.version; let i = $index; let last = $last; let first = $first){
                     <mat-list-item
-                        *ngFor="let event of data?.events; let i = index"
+                        *ngIf="first || !last || !event.hasNext"
```
Comment (sergeysenja1992) [inline packages/administration/dashboards-config/configuration-history/config-history-modal/config-history-modal.component.html:54]: maybe if also can be replaced by @if

File: packages/administration/dashboards-config/configuration-history/config-history-modal/config-history-modal.component.ts [hunk 3]
```diff
@@ -70,12 +70,20 @@ export class ConfigHistoryModalComponent implements OnInit {
...
-        return event === this.data?.events?.[0];
+        return event.date === this.data?.events?.[0]?.date;
     }
 
     private setPrevValues(index: number): void {
         const prevEvent = this.data?.events?.[index + 1];
         this.prevConfig = prevEvent?.config ?? '{}';
         this.prevDate = prevEvent?.date ?? null;
     }
+    public loadMore(): void {
```
[the remark this answers is not in the document]
Comment (Bozonych) [reply packages/administration/dashboards-config/configuration-history/config-history-modal/config-history-modal.component.ts:81]: fixed

File: packages/administration/dashboards-config/injectors.ts (hunk 1 above)
Comment (Bozonych) [reply packages/administration/dashboards-config/injectors.ts:25]: fixed

File: packages/administration/dashboards-config/configuration-history/config-history-modal/config-history-modal.component.html (hunk 2 above)
Comment (Bozonych) [reply packages/administration/dashboards-config/configuration-history/config-history-modal/config-history-modal.component.html]: Migrate all template to control flow.
P.S use for this `ng g @angular/core:control-flow-migration` command, itʼs amazing

**Review by Olensavchenko — APPROVED** [review]

**Review by LevBD — APPROVED** [review]

**Pull request merged**